platform/kube/controller_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kube
+
+import (
+	"testing"
+
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/pkg/util/intstr"
+)
+
+func TestIngressIgnored(t *testing.T) {
+	cases := []struct {
+		ingressClass string
+		shouldIgnore bool
+	}{
+		{ingressClass: "nginx", shouldIgnore: true},
```

Comment:
We cannot ensure that others are well-behaving, but we can ensure we are.
For what it's worth, GKE and Nginx (both community and Nginx Inc) ingress controllers respect it.